internal/configuration/configuration.go

```diff
@@ -75,37 +77,41 @@ func (m *Manager) GetWorkloads() models.WorkloadList {
 	return m.deviceConfiguration.Workloads
 }
 
-func (m *Manager) Update(message models.DeviceConfigurationMessage) error {
+func (m *Manager) Update(message models.DeviceConfigurationMessage) []error {
 	configurationEqual := reflect.DeepEqual(message.Configuration, m.deviceConfiguration.Configuration)
 	workloadsEqual := reflect.DeepEqual(message.Workloads, m.deviceConfiguration.Workloads)
 	log.Tracef("Initial config: [%v]; workloads equal: [%v]; configurationEqual: [%v]", m.IsInitialConfig(), workloadsEqual, configurationEqual)
+	errors := []error{}
 	if m.IsInitialConfig() || !(configurationEqual && workloadsEqual) {
 		log.Tracef("Updating configuration: %v", message)
 		for _, observer := range m.observers {
 			err := observer.Update(message)
 			if err != nil {
-				return err
+				errors = append(errors, fmt.Errorf("Cannot update observer, err: %s", err))
+				return errors
 			}
 		}
 
 		// TODO: handle all the failure scenarios correctly; i.e. compensate all the changes that has already been introduces.
 		file, err := json.MarshalIndent(message, "", " ")
 		if err != nil {
-			return err
+			errors = append(errors, fmt.Errorf("Cannot unmarshal Json, err: %s", err))
```

Comment:
s/Json/JSON
You can find more here: https://github.com/golang/go/wiki/CodeReviewComments#initialisms
Also, no need to mention ", err:", just a colon will do :smile: 
```
fmt.Errorf("cannot unmarshal JSON: %s", err)
```
WDYT?